catalog/compose.yml

```diff
@@ -7,7 +7,7 @@ x-airflow-common: &airflow-common
   depends_on:
     - postgres
     - s3
-  image: openverse_catalog
+  image: openverse-catalog
```

Comment:
Why does this `image` entry exist at all on these? Every time I run `just up` these days, docker tries to retrieve these images and it fails with a (seemingly) random authentication error. What's the purpose of these image names on our services, exactly? If they aren't necessary, I'd much rather remove them from everywhere.

Reply:
Having both [`image` and `build` in the Compose file](https://docs.docker.com/compose/compose-file/build/#using-build-and-image) means that Docker will first try to pull it and if the image is not found, it will be built. This is standard and expected behaviour so I am wondering if what you're seeing is an info/ warning message and not an error.

Also, we can theoretically control the pulling behaviour using [`pull_policy`](https://docs.docker.com/compose/compose-file/05-services/#pull_policy). I suppose setting the `pull_policy` to "never" might get rid of that message.

Reply:
It is a warning log, not an error (I misspoke). I suppose it is recent behaviour that it actually tries to pull the images, or at least that it warns me that it wasn't able to. Anecdotally, it sometimes spends several seconds to a minute trying to pull the images. I suppose if someone published an image by these names to `docker.io` or whatever default registry a given container runtime uses it would actually try to pull that instead of building locally. That doesn't seem right? In what cases do any of these `image` entries do anything other than waste cycles trying to download images that aren't findable?

> I suppose setting the pull_policy to "never" might get rid of that message.

Why would that be different than just removing `image` altogether?

Does `image` tell Docker to try to reuse the _local_ image? And if so, why is that better than just the build instructions? If the local image is present, then Docker will use the cache, and it wouldn't spend nearly any time at all trying to build. If `pull_policy: never` combines the two (never pull, but still try to find it locally), then I guess that _is_ the best of both worlds. I'm surprised the image entry is needed at all for images that are nearly always locally built (in any meaningful case) :thinking:

Reply:
I think what `image` does for us is allow us to use a name for the image that's different from the service name (otherwise Docker will automatically tag the image based on the name of the service, which is not always what we want).

It also allows us to reuse a image for multiple services as once an image has been built for a service, other services can directly use it without building copies themselves.

Reply:
Could we simplify all this by changing the name of the services, as recommended in #1910?

Does docker cache using the Dockerfile and context, so examples like the catalog's three services would still have the same images? Maybe that's wrong. I'll try it out tomorrow if you don't have a definitive answer.

It sounds like `pull_policy: never` would solve the warning, though? If not get rid of the image names?

Reply:
Yes, simplifying the names of services will remove some of these `image` fields. Not all, but at least those where the service name is the same as that of the image.

For services with different names, different images will be created by Docker. Because of efficiency, they will share underlying images and layers, so space consumed will not go up, but it will still lead to multiple image tags.

`pull_policy: never` is definitely useful to avoid pulling a random, potentially malicious image from Docker, and will solve that warning. Should that be part of this PR?

Reply:
This PR is fine as is to me, I was just wanting to know if there's a bigger change that helps address some other issues :slightly_smiling_face: However you feel is the best way to solve these devex problems is good to me! I suppose the only thing to keep in mind is that #1910 will affect how people run just scripts, just logs, in particular being one I use a lot and relies on the service names, but I always mess up and write `just logs api` anyway LOL. That just means it would need to be communicated more clearly than changes like this that don't affect how people interact with it.

But really, from my perspective, up to you how you think is the best way to solve these issues :+1: Aside from the specific details, whether in this PR, another one, whatever works!